api/src/services/sample-location-service.ts

```diff
@@ -64,13 +64,24 @@ export class SampleLocationService extends DBService {
    */
   async insertSampleLocations(sampleLocations: PostSampleLocations): Promise<SampleLocationRecord[]> {
     const methodService = new SampleMethodService(this.connection);
-
+    const determineName = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const nameKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'name' || key.toLowerCase() === 'label'
+      );
+      return nameKey && geometry.properties ? geometry.properties[nameKey].substring(0, 50) : '';
+    };
+    const determineDesc = (geometry: Feature<Geometry | GeometryCollection, Properties>) => {
+      const descKey = Object.keys(geometry.properties ?? {}).find(
+        (key) => key.toLowerCase() === 'desc' || key.toLowerCase() === 'description'
+      );
+      return descKey && geometry.properties ? geometry.properties[descKey].substring(0, 250) : '';
```

Comment:
If I understand this right: `determineName` and `determineDesc` will try to get the names and descriptions from the features themselves. If these properties don't exist, then empty string is returned. Then when they are being inserted in the database, Postgres will coerce to the default value (Sample Site #n). Do I have that right?

Reply:
Yeah, that's right. Perhaps all that logic should be in the same place, since I wound up getting pretty confused at first about why the name field was being overridden from what I had already set here. I could make a separate query function in the repository file to call in here, I imagine it's not especially inefficient compared to leaving that coercion as a subquery.

Reply:
Your current approach isn't bad. But I would remove the `determineName` function (since it's just called immedately after definition) to something like
```javascript
const shapeFileFeatureName: string | undefined = ...
```

This way, we're explicitly declaring that the shapefile feature might not have a name, so when we coerce it later, it makes more sense.